metric_learn/nca.py

```diff
@@ -1,6 +1,40 @@
 """
 Neighborhood Components Analysis (NCA)
 Ported to Python from https://github.com/vomjom/nca
+
+Neighborhood Components Analysis (`NCA`) is a distance
```

Comment:
Now that we're at it, I know that it's not completely done in `metric-learn` yet, but we should try to adopt scikit-learn conventions for the user guide, i.e. : 
- We should add short descriptions of the algorithms (without mathematical formulation) in the docstring of the class we want to document
- We should put the full description in the user guide 

So I think what you could do is : 

- put a reference at the top of `metric_learn.nca.rst` like this: 
https://github.com/scikit-learn/scikit-learn/blame/4140657700cc55830347c871134c8e982d29fab5/doc/modules/neighbors.rst#L515-L517

- Put the following description in the docstring of the class `NCA` (at the top of the file we should just keep a title, the authors, and the encoding) with this line as a reference to the user guide: https://github.com/scikit-learn/scikit-learn/blob/4140657700cc55830347c871134c8e982d29fab5/sklearn/neighbors/nca.py#L37

Reply:
Cool, I should do it for all the docstrings..I guess?